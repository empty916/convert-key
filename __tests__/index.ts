import createConvertUtil from "../src/index";

const keyMaps = {
  a: "A1",
  b: "B1",
  f: "F1",
} as const; // keyMaps Object must be delared as a const

const data = {
  a: 1,
  b: {
    b1: 1,
    b2: {},
    b3: "1",
  },
  c: null,
  d: "",
  // f: {}
};

test("create", () => {
    // @ts-ignore
    expect(() => createConvertUtil(null)).toThrow();

    const {convert, revert} = createConvertUtil(keyMaps);

    // @ts-ignore
    expect(() => convert(null)).toThrow();
    // @ts-ignore
    expect(() => convert(1)).toThrow();
    // @ts-ignore
    expect(() => convert('null')).toThrow();
    
    // @ts-ignore
    // @ts-ignore
    expect(() => revert(null)).toThrow();
    // @ts-ignore
    expect(() => revert(1)).toThrow();
    // @ts-ignore
    expect(() => revert('null')).toThrow();
});

test("convert", () => {
  const { convert } = createConvertUtil(keyMaps);
  expect(convert(data)).toMatchSnapshot();
});

test("revert", () => {
  const { convert, revert } = createConvertUtil(keyMaps);
  expect(revert(convert(data))).toEqual(data);
});
