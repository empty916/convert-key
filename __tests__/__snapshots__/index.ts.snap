// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert 1`] = `
{
  "A1": 1,
  "B1": {
    "b1": 1,
    "b2": {},
    "b3": "1",
  },
  "c": null,
  "d": "",
}
`;
